app/src/main/java/org/p2p/wallet/home/ui/main/delegates/token/TokenDelegate.kt

```diff
@@ -0,0 +1,86 @@
+package org.p2p.wallet.home.ui.main.delegates.token
+
+import androidx.core.view.isVisible
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.hannesdorfmann.adapterdelegates4.AdapterDelegate
+import com.hannesdorfmann.adapterdelegates4.dsl.AdapterDelegateViewBindingViewHolder
+import com.hannesdorfmann.adapterdelegates4.dsl.adapterDelegateViewBinding
+import org.p2p.core.glide.GlideManager
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.wallet.common.ui.recycler.swipe.SwipeRevealLayout
+import org.p2p.wallet.databinding.ItemTokenBinding
+import org.p2p.wallet.home.ui.main.adapter.TokenAdapter
+import org.p2p.wallet.utils.withTextOrGone
+
+private const val VIEW_ALPHA_MAX_VALUE = 0.8f
+
+private typealias BindListener = ((binding: ItemTokenBinding, item: TokenCellModel) -> Unit)
+
+private val inflateViewBinding = { inflater: LayoutInflater, parent: ViewGroup ->
+    ItemTokenBinding.inflate(inflater, parent, false)
+}
+
+fun tokenDelegate(
+    glideManager: GlideManager,
+    onBindListener: BindListener? = null,
+): AdapterDelegate<List<AnyCellItem>> =
+    adapterDelegateViewBinding<TokenCellModel, AnyCellItem, ItemTokenBinding>(
+        viewBinding = inflateViewBinding
+    ) {
+        bind { payloads ->
+            @Suppress("UNCHECKED_CAST")
+            val fields = payloads.firstOrNull() as? Set<String>
+
+            if (fields.isNullOrEmpty()) {
+                onBind(glideManager)
+            } else {
+                fields.forEach { field ->
+                    when (field) {
+                        TokenAdapter.DIFF_FIELD_TOKEN_BALANCE -> bindBalance()
+                    }
+                }
+            }
```

Comment:
fields.forEach {  
   when (field) {

can be extracted to a method i think. smth like 
onBindWithDiff
onDiffBind